kaldb/src/main/java/com/slack/kaldb/logstore/schema/SchemaAwareLogDocumentBuilderImpl.java

```diff
@@ -347,31 +358,148 @@ public static SchemaAwareLogDocumentBuilderImpl build(
+      addField(doc, LogMessage.ReservedField.NAME.fieldName, message.getName(), "", 0);
+    }
+    if (message.getDuration() != 0) {
+      jsonMap.put(
+          LogMessage.ReservedField.DURATION_MS.fieldName,
+          Duration.of(message.getDuration(), ChronoUnit.MICROS).toMillis());
+      addField(
+          doc,
+          LogMessage.ReservedField.DURATION_MS.fieldName,
+          Duration.of(message.getDuration(), ChronoUnit.MICROS).toMillis(),
+          "",
+          0);
+    }
+    if (!message.getId().isEmpty()) {
+      jsonMap.put(LogMessage.SystemField.ID.fieldName, message.getId().toStringUtf8());
+      addField(doc, LogMessage.SystemField.ID.fieldName, message.getId().toStringUtf8(), "", 0);
+    } else {
+      throw new IllegalArgumentException("Span id is empty");
+    }
+
+    // this interferes in tests
+    // Instant timestamp = SpanFormatter.getTimestampFromSpan(message);
+    Instant timestamp =
+        Instant.ofEpochMilli(
+            TimeUnit.MILLISECONDS.convert(message.getTimestamp(), TimeUnit.MICROSECONDS));
     addField(
-        doc,
-        LogMessage.SystemField.TIME_SINCE_EPOCH.fieldName,
-        message.getTimestamp().toEpochMilli(),
-        "",
-        0);
-    addField(doc, LogMessage.ReservedField.TYPE.fieldName, message.getType(), "", 0);
-    addField(doc, LogMessage.SystemField.ID.fieldName, message.getId(), "", 0);
-
-    final String msgString = JsonUtil.writeAsString(message.toWireMessage());
+        doc, LogMessage.SystemField.TIME_SINCE_EPOCH.fieldName, timestamp.toEpochMilli(), "", 0);
+
+    Map<String, Trace.KeyValue> tags =
+        message.getTagsList().stream()
+            .map(keyValue -> Map.entry(keyValue.getKey(), keyValue))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    // TODO: this should just be top level Trace.Span fields. This is error prone - what if type is
+    // not a string?
+    String indexName =
+        tags.containsKey(LogMessage.ReservedField.SERVICE_NAME.fieldName)
+            ? tags.get(LogMessage.ReservedField.SERVICE_NAME.fieldName).getVStr()
+            : DEFAULT_INDEX_NAME;
+    String msgType =
+        tags.containsKey(LogMessage.ReservedField.TYPE.fieldName)
+            ? tags.get(LogMessage.ReservedField.TYPE.fieldName).getVStr()
+            : DEFAULT_LOG_MESSAGE_TYPE;
+
+    jsonMap.put(LogMessage.ReservedField.TYPE.fieldName, msgType);
+    addField(doc, LogMessage.ReservedField.TYPE.fieldName, msgType, "", 0);
+
+    jsonMap.put(LogMessage.ReservedField.SERVICE_NAME.fieldName, indexName);
+    addField(doc, LogMessage.SystemField.INDEX.fieldName, indexName, "", 0);
+    addField(doc, LogMessage.ReservedField.SERVICE_NAME.fieldName, indexName, "", 0);
+
+    tags.remove(LogMessage.ReservedField.SERVICE_NAME.fieldName);
+    tags.remove(LogMessage.ReservedField.TYPE.fieldName);
+
+    for (Trace.KeyValue keyValue : tags.values()) {
```

Comment:
Since we don't use int32 and float32 types, why did we add them to the proto definition?

Reply:
Since Lucene only supports 64 bit int's why do we need 32 int's the type definition? It may be better to keep the types simple and just support 64bit ints for now.